### PR TITLE
Add numeric functions

### PR DESCRIPTION
**Pull request by peterdemartini**
- [x] Ensure that the data frame adapter can properly validate the accepts
- [x] Add examples for `add` and `subtract`
- [x] Move the current `divide` to `divideValues` and make divide act more like `add` and `subtract`
- [x] Add and fix `inNumberRange` to with bigints
- [x] Add support for `aliases` and ensure there are no duplicate names
- [x] Fix `joinList` to work with readonly arrays
- [x] Add `addValues` and `subtractValues` to stay in the same group as divide
- [x] Add `isGreaterThan`
- [x] Add `isGreaterThanOrEqual`
- [x] Add `isLessThan`
- [x] Add `isLessThanOrEqual`
- [x] Add `modulus`
- [x] Add `toPrecision`
- [x] Add `isEven`
- [x] Add `isOdd`
- [x] Add `toCelsius`
- [x] Add `toFahrenheit`

### CODE REVIEW

**Review by jsnoble — APPROVED** [review]
LGTM

File: packages/data-mate/src/function-configs/interfaces.ts [hunk 1]
```diff
@@ -70,38 +70,51 @@ export interface FunctionDefinitionExample<T extends Record<string, any>> {
 
 export interface FunctionDefinitionConfig<T extends Record<string, any>> {
     /**
-     * The name of the function
+     * The name of the function, this should be consider case-insensitive
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/interfaces.ts:73]: typo
`considered case-insensitive` or just `this should be case-insensitive`

File: packages/data-mate/src/function-configs/numeric/add.ts [hunk 2]
```diff
@@ -8,51 +8,90 @@ import {
...
-        FieldType.Long,
-        FieldType.Double
     ],
     argument_schema: {
-        by: {
+        value: {
             type: FieldType.Number,
             array: false,
-            description: 'How much to add, defaults to 1'
+            description: 'How much to add'
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/add.ts:81]: this sounds weird to me...maybe `value that will be added to the input`?

File: packages/data-mate/src/function-configs/numeric/inNumberRange.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    FunctionDefinitionType,
+    FunctionDefinitionCategory,
+} from '../interfaces';
+
+export const inNumberRangeConfig: FieldValidateConfig<InNumberRangeArg> = {
+    name: 'inNumberRange',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    category: FunctionDefinitionCategory.NUMERIC,
+    description: 'Check to see if a number exists within a given min and max value, this can configured to be inclusive or exclusive',
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/inNumberRange.ts:15]: consider making the description less wordy.

`Checks if a number is within a given min and max value, option for inclusive or exclusive`

File: packages/data-mate/src/function-configs/numeric/isEven.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    FunctionDefinitionType,
+    FunctionDefinitionCategory,
+} from '../interfaces';
+
+export const isEvenConfig: FieldValidateConfig = {
+    name: 'isEven',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    category: FunctionDefinitionCategory.NUMERIC,
+    description: 'Check to see if a number is even',
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/isEven.ts:14]: Consider ` Check if a number is even`

File: packages/data-mate/src/function-configs/numeric/subtract.ts [hunk 5]
```diff
@@ -8,51 +8,89 @@ import {
...
-        FieldType.Float,
-        FieldType.Long,
-        FieldType.Double
     ],
     argument_schema: {
-        by: {
+        value: {
             type: FieldType.Number,
             array: false,
             description: 'How much to subtract, defaults to 1'
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/subtract.ts:81]: `value to subtract` ?

File: packages/data-mate/src/function-configs/numeric/isGreaterThan.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+export interface GreaterThanArgs {
+    readonly value: number;
+}
+
+export const isGreaterThanConfig: FieldValidateConfig<GreaterThanArgs> = {
+    name: 'isGreaterThan',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    category: FunctionDefinitionCategory.NUMERIC,
+    description: 'Check to see if a number is greater than the specified value',
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/isGreaterThan.ts:18]: could be less wordy, `Checks if a number...`

File: packages/data-mate/src/function-configs/numeric/isGreaterThanOrEqual.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+export interface GreaterThanOrEqualToArgs {
+    readonly value: number;
+}
+
+export const isGreaterThanOrEqualToConfig: FieldValidateConfig<GreaterThanOrEqualToArgs> = {
+    name: 'isGreaterThanOrEqualTo',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    category: FunctionDefinitionCategory.NUMERIC,
+    description: 'Check to see if a number is greater than or equal to the specified value',
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/isGreaterThanOrEqual.ts:18]: could be less wordy, Checks if a number...

File: packages/data-mate/src/function-configs/numeric/isLessThan.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+export interface LessThanArgs {
+    readonly value: number;
+}
+
+export const isLessThanConfig: FieldValidateConfig<LessThanArgs> = {
+    name: 'isLessThan',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    category: FunctionDefinitionCategory.NUMERIC,
+    description: 'Check to see if a number is less than the specified value',
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/isLessThan.ts:18]: could be less wordy, `Checks if a number...`

File: packages/data-mate/src/function-configs/numeric/isLessThanOrEqual.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+export interface LessThanOrEqualToArgs {
+    readonly value: number;
+}
+
+export const isLessThanOrEqualToConfig: FieldValidateConfig<LessThanOrEqualToArgs> = {
+    name: 'isLessThanOrEqualTo',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    category: FunctionDefinitionCategory.NUMERIC,
+    description: 'Check to see if a number is less than or equal to the specified value',
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/isLessThanOrEqual.ts:18]: could be less wordy, `Checks if a number...`

File: packages/data-mate/src/function-configs/numeric/isOdd.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    FunctionDefinitionType,
+    FunctionDefinitionCategory,
+} from '../interfaces';
+
+export const isOddConfig: FieldValidateConfig = {
+    name: 'isOdd',
+    type: FunctionDefinitionType.FIELD_VALIDATION,
+    process_mode: ProcessMode.INDIVIDUAL_VALUES,
+    category: FunctionDefinitionCategory.NUMERIC,
+    description: 'Check to see if a number is even',
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/numeric/isOdd.ts:14]: could be less wordy, `Checks if a number...`

File: packages/data-mate/src/function-configs/numeric/add.ts (hunk 2 above)
Comment (ciorg) [reply packages/data-mate/src/function-configs/numeric/add.ts:81]: or `value added to input`

**Review by ciorg — APPROVED** [review]

**Pull request merged**